src/Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <vector>
+#include <Core/NamesAndTypes.h>
+#include <Core/Range.h>
+
+namespace Iceberg
+{
+
+enum class PartitionTransform
+{
+    Year,
+    Month,
+    Day,
+    Hour,
+    Identity,
+    Void,
+    Unsupported
+};
+
+struct SpecificSchemaPartitionInfo
+{
+    std::vector<std::vector<DB::Range>> ranges;
+    DB::NamesAndTypesList partition_names_and_types;
+};
+
+Iceberg::PartitionTransform getTransform(const String & transform_name);
+
+DateLUTImpl::Values getValues(Int32 value, Iceberg::PartitionTransform transform);
+
+Int64 getTime(Int32 value, Iceberg::PartitionTransform transform);
+
+Int16 getDay(Int32 value, Iceberg::PartitionTransform transform);
```

Comment:
It seems like these 3 functions are used only inside `PartitionPrunning.cpp`, so there is no need to declare them in `.h` file.
Also their names are too broad, it's not clear what they do from their names